lib/core/stores/remote.ts

```diff
@@ -8,76 +8,102 @@ import {
   isDurablePromise,
   isCompletedPromise,
 } from "../promise";
-import { IPromiseStore, isSearchPromiseResult } from "../store";
+import { IPromiseStore, IScheduleStore, IStore, isSearchPromiseResult, isSearchSchedulesResult } from "../store";
 import { IEncoder } from "../encoder";
 import { Base64Encoder } from "../encoders/base64";
 import { ErrorCodes, ResonateError } from "../error";
 import { ILogger } from "../logger";
+import { Schedule, isSchedule } from "../schedule";
+
+async function call<T>(
+  url: string,
+  guard: (b: unknown) => b is T,
+  options: RequestInit,
+  retries: number = 3,
```

Comment:
Can we make this the last argument? That way we won't have to specify it on each call.